src/pickit5.c

```diff
@@ -627,7 +627,72 @@ static int pickit5_open(PROGRAMMER *pgm, const char *port) {
       my.pgm_type = PGM_TYPE_SNAP;
   }
 
-  return rv;
+  // No PICkit4 / SNAP (PIC mode) nor PICkit5 found, look for programmer in AVR mode
+  if(rv >= 0)  // if a programmer in PIC mode found, we're done
+    return rv;
+
+  // otherwise, try to see if there is a SNAP or PK4 in AVR mode
+  pinfo.usbinfo.vid = USB_VENDOR_ATMEL;
+  pinfo.usbinfo.pid = USB_DEVICE_SNAP_AVR_MODE;
+
+  pgm->fd.usb.max_xfer = USBDEV_MAX_XFER_3;
+  pgm->fd.usb.rep = USBDEV_BULK_EP_READ_3;
+  pgm->fd.usb.wep = USBDEV_BULK_EP_WRITE_3;
+  pgm->fd.usb.eep = USBDEV_EVT_EP_READ_3;
+
+  const char *pgm_suffix = strchr(pgmid, '_')? strchr(pgmid, '_'): "";
+  char part_option[32] = {};  // 32 should be enough
+
+  if(partdesc)
+    snprintf(part_option, 31, "-p %s ", partdesc);
```

Comment:
Best to replace 31 with `sizeof part_option`. There is no need to subtract 1 as `snprintf` writes at most $n$ bytes *and* nul-terminates the string.